algorithms/src/std_algorithms/impl/Kokkos_IsSortedUntil.hpp

```diff
@@ -28,33 +28,30 @@ namespace Kokkos {
 namespace Experimental {
 namespace Impl {
 
-template <class IteratorType, class IndicatorViewType, class ComparatorType>
+template <class IteratorType, class ComparatorType, class ReducerType>
 struct StdIsSortedUntilFunctor {
-  using index_type = typename IteratorType::difference_type;
+  using index_type     = typename IteratorType::difference_type;
+  using red_value_type = typename ReducerType::value_type;
```

Comment:
I'm not a big fan of abbreviations that sound like colors. 🙂 Also, `FunctorAnalysis` would try using a `value_type` alias instead of deducing it from the functors join operator but that doesn't quite apply here since we have a reducer anyway.
```suggestion
  using value_type = typename ReducerType::value_type;
```
Feel free to ignore.

Reply:
next time i will use animal names :)